Userland/Applications/CertificateSettings/CertificateStore.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, Fabian Dellwing <fabian@dellwing.net>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibGUI/Model.h>
+#include <LibGUI/SettingsWindow.h>
+#include <LibGUI/SortingProxyModel.h>
+#include <LibGUI/TableView.h>
+#include <LibTLS/Certificate.h>
+
+namespace CertificateSettings {
+
+class CertificateStoreProxyModel : public GUI::SortingProxyModel {
+public:
+    static ErrorOr<NonnullRefPtr<CertificateStoreProxyModel>> create(NonnullRefPtr<Model> source, NonnullRefPtr<GUI::TableView> view)
+    {
+        return adopt_nonnull_ref_or_enomem(new (nothrow) CertificateStoreProxyModel(move(source), move(view)));
+    }
+    virtual void sort(int column, GUI::SortOrder) override;
+
+private:
+    CertificateStoreProxyModel(NonnullRefPtr<Model> source, NonnullRefPtr<GUI::TableView> view)
+        : SortingProxyModel(move(source))
+        , m_parent_table_view(move(view)) {};
+
+    NonnullRefPtr<GUI::TableView> m_parent_table_view;
+};
+
+class CertificateStoreModel : public GUI::Model {
+public:
+    enum Column {
+        IssuedTo,
+        IssuedBy,
+        Expire,
+        __Count
+    };
+
+    static NonnullRefPtr<CertificateStoreModel> create();
+    virtual ~CertificateStoreModel() override = default;
+
+    virtual int row_count(GUI::ModelIndex const& = GUI::ModelIndex()) const override { return m_certificates.size(); }
+    virtual int column_count(GUI::ModelIndex const& = GUI::ModelIndex()) const override { return Column::__Count; }
+    virtual DeprecatedString column_name(int) const override;
+    virtual GUI::Variant data(GUI::ModelIndex const&, GUI::ModelRole) const override;
+    virtual ErrorOr<void> load();
+    virtual ErrorOr<size_t> try_add(Vector<Certificate> const&);
+    virtual Certificate get(int index);
+
+private:
+    Vector<Certificate> m_certificates;
+};
+
+class CertificateStoreWidget : public GUI::SettingsWindow::Tab {
+    C_OBJECT_ABSTRACT(CertStoreWidget)
+public:
+    virtual ~CertificateStoreWidget() override = default;
+    static ErrorOr<NonnullRefPtr<CertificateStoreWidget>> try_create();
+    virtual void apply_settings() override {};
+
+private:
+    CertificateStoreWidget() = default;
+    ErrorOr<void> initialize();
+    ErrorOr<void> try_import();
+    ErrorOr<void> try_export();
```

Comment:
probably no need to use `try_` prefix here (i.e. in new code; same for `try_add`).

Reply:
I like to see on first look if I need to handle errors on invoking a function. So I opted into the `try_` style.

Reply:
We only prefix function names with `try_` when there is an infallible counterpart that we are adding a fallible variant for. We don't use it for new code like this, as it makes the callsite needlessly verbose (i.e. `TRY(try_..))`)

Reply:
Ok, will rename them 👍

Reply:
As `export` is reserved, I choose `import_pem` and `export_pem`.